### ISSUE TITLE
Mizselect widget with mizdb-inlines formsets 

### ISSUE THREAD

**Issue by sasikumar1978k** [Actionb/mizdb-tomselect#12]
I have one issue in formset , widget MIZSelect is not  working in added form of formsets. And filter_by option is not working in formsets forms properly. How can I use custom fieldname with Search_lookup

**Comment by sasikumar1978k** [comment]
Sorry for the confusion Mizselect widget working fine in mizinline formset . Filter option is working but small issue is there . when i select first field the dependent field should be set to none then only can choose right option from second field. here is the issue 
![Screenshot from 2023-09-29 11-06-44](https://github.com/Actionb/mizdb-tomselect/assets/23027374/fb385fe3-e20c-4056-aab8-d62a1051a6d8)
![Screenshot from 2023-09-29 11-07-00](https://github.com/Actionb/mizdb-tomselect/assets/23027374/4fe9cf5d-2652-4477-a2d6-cfc3fd880e97)
![Screenshot from 2023-09-29 11-07-17](https://github.com/Actionb/mizdb-tomselect/assets/23027374/4977c3b3-f000-4d50-be5c-125791c332c5)
![Screenshot from 2023-09-29 11-07-29](https://github.com/Actionb/mizdb-tomselect/assets/23027374/c2916be8-0b6a-4d2d-9e70-875c832f6913)

**Comment by sasikumar1978k** [comment]
search_lookup option is not working if model don't have 'name' field

    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'name' into field. Choices are: id, languageknownst, longuagename, staffdet

my code:
 self.fields['language'] = forms.ModelChoiceField(
        languages.objects.all(),
        widget=MIZSelect(languages,
                         search_lookup= f"longuagename__icontains", 
                         url='mizdb_autocomplete_view',can_remove=True),
    )

Model:
class languages(models.Model):
    longuagename = models.CharField(max_length=50,default='', unique = True  , verbose_name= 'Language')


![Screenshot from 2023-09-29 11-16-54](https://github.com/Actionb/mizdb-tomselect/assets/23027374/f54343a8-cdcf-446f-a1c1-018f63627e94)

**Comment by Actionb** [comment]
Are you sure that `Degreesubject` is actually filtered by `Degree`? Can I see the code where you declare the formsets?

When the filter by element changes, the event handler drops all options of the select element that are not selected:
https://github.com/Actionb/mizdb-tomselect/blob/e930e98ccdf77295780dd075bcbcee8a776d032a/client/mizselect.js#L85C1-L99C4

**Comment by sasikumar1978k** [comment]
**Yes**  Degreesubject is actually filtered by Degree.

Degree subjects are not same for both degrees  . For example Botany not in MCA degree 

form.py

class QualificationForm(forms.ModelForm):
    def __init__(self, *args, **kwargs):
        super(QualificationForm, self).__init__(*args, **kwargs)
        
        self.fields['degree'] = forms.ModelChoiceField(
        queryset=degree.objects.all().order_by('name'),
        widget=MIZSelect(degree,
                         url='mizdb_autocomplete_view'
                         )
        
            )
        
        self.fields['degreesubject'] = forms.ModelChoiceField(
        queryset=degreesub.objects.all().order_by('name'),
        widget=MIZSelect(degreesub,
                         filter_by=("degree", "degree"),
                            url='mizdb_autocomplete_view'
                           )
           
            )

        
    class Meta:
        model = staffqualification
        fields = '__all__'
        widgets = {
            
            "course_admn_date": DatePickerInput(),
            "course_completion_date": DatePickerInput(),
            "pass_date": DatePickerInput(),

          
        }


model.py

class degree(models.Model):
    name = models.CharField(max_length=50,default='', verbose_name='Name of the Degree')
    type = models.ForeignKey(degreetype, on_delete=models.CASCADE, null=True)
     
    def __str__(self):
        return self.name
    def save(self, *args, **kwargs):
        for field_name in ['name',]:
            val = getattr(self, field_name, False)
            if val:
                setattr(self, field_name, val.upper())
        super(degree, self).save(*args, **kwargs)
    class Meta:
        unique_together = ('name', 'type',)

class degreesub(models.Model):
    degree = models.ForeignKey(degree, on_delete=models.CASCADE, null=True)
    name = models.CharField(max_length=150,default='',verbose_name='Degree Subject')
   
    def __str__(self):
        return self.name
    
    def save(self, *args, **kwargs):
        for field_name in ['name',]:
            val = getattr(self, field_name, False)
            if val:
                setattr(self, field_name, val.upper())
        super(degreesub, self).save(*args, **kwargs)
    class Meta:
        unique_together = ('degree', 'name',)

**Comment by Actionb** [comment]
## Explanation
When changing the value of the filter by element, the options for currently or previously selected items are not removed from the list of options. So if you select BSC as degree and Botany as subject, and then switch degree to MCA, then Botany will still be shown in the list of options, because it is or was selected.

## Reproducing the scenario in your screenshot
I assume you did this:
1. select MCA
2. select Computer Science
3. select BSC 
4. remove Computer Science
5. open dropdown 

The list of options now contains: Computer Science as first item (because it was previously selected) and Botany (because it satisfies the filter).

And proceeding:
6. select Botany
7. select MCA

The list of options now contains: Botany (selected item) and Computer Science (satisfies filter).

## To summarize
 This is working as expected and not a bug. But perhaps it could be made more clear why the option for the selected item persists in the list of options after changing the filter element.
